src/probability_theory/notation.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import measure_theory.function.conditional_expectation
+import measure_theory.decomposition.radon_nikodym
+
+/-! # Notations for probability theory -/
+
+open measure_theory measure_theory.measure topological_space
+
+-- The related notation `ℙ[ X | hm] := measure_theory.condexp hm ℙ X` is defined in
+-- measure_theory.function.conditional_expectation.
+localized "notation `𝔼[` X `|` hm `]` := measure_theory.condexp hm volume X" in probability_theory
+
+-- The usual expectation notation `𝔼[X]` does not carry information about the measure used, hence
+-- we reserve it for the `volume` measure, and use the similar `ℙ[X]` for the expectation under `ℙ`.
+localized "notation ℙ `[` X `]` := ∫ x, X x ∂ℙ" in probability_theory
+
+localized "notation `𝔼[` X `]` := ∫ a, X a" in probability_theory
+
+localized "notation X `=ₐₛ`:50 Y:50 := X =ᵐ[volume] Y" in probability_theory
+
+localized "notation X `≤ₐₛ`:50 Y:50 := X ≤ᵐ[volume] Y" in probability_theory
+
+localized "notation `∂` ℙ `/∂`:50 ℙ':50 := ℙ.rn_deriv ℙ'" in probability_theory
+
+section examples
+
+open_locale probability_theory
+
+variables {α E : Type*} [measure_space α] {ℙ ℙ' : measure α} [measurable_space E] [normed_group E]
+  [normed_space ℝ E] [borel_space E] [second_countable_topology E] [complete_space E] {X Y : α → E}
+
+example : ℙ[X] = ∫ a, X a ∂ℙ := rfl
```

Comment:
I don't really like this one. I'd prefer `ℙ` to be a notation for `volume`, and then `ℙ s` would be the probability of the set `s`. Or maybe rather for the real-valued version of the measure (do we already have an API around this for finite measures?)

Reply:
You're right that `ℙ s` should be the probability of a set. I should not use `ℙ` as a name for a measure.
But it does not rule out `P[X]` where `P` is the measure, and not part of the notation.

I really want a way to have an expectation with respect to a non-volume measure, since almost every paper I write uses a change of probability measure somewhere.

Reply:
The `P[X]` notation is actually used by some people to denote the expected value under `P`, and the other alternative I saw in real math is `𝔼_P[X]` (with a subscript P), which is not something we can do.

Reply:
You're right that `ℙ s` should be the probability of a set, either as notation for `volume s` or as a consequence of your proposal of `ℙ` being a notation for the volume. In any case, I should not use `ℙ` as a name for a measure.

But it does not rule out `µ[X]` where `µ` is the measure, and not part of the notation. I think those two notations could go well together.
If `ℙ` is a notation for `volume` as you suggest, then with the `µ[X]` notation for expected value, `ℙ[X]` would be a way to write the expected value of `X`. Or we can still define the `𝔼[X]` notation on top of that if we prefer.

I really want a nice way to write an expectation with respect to a non-volume measure, since almost every paper I write uses a change of probability measure somewhere (I realize that this contradicts the first sentence of the PR description, in which I say that the measure is often implicit).

The notations I see most often in real math to make the measure explicit are `𝔼_P[X]` or `𝔼_{X∼P}[X]` (with subscripts), which is not something we can do. The `P[X]` notation is also something I've seen used by some people to denote the expected value under `P`, and that one is mathlib-compatible.

Reply:
I am perfectly ok with `µ[X]`, although maybe `𝔼[X; µ]` would also work. I'll let you choose whichever you prefer.

Reply:
Regarding the real vs ennreal issue for `ℙ s`, I think it could be nice to have it real because working with ennreal is not fun.
But that means that `ℙ` cannot be a notation for the volume. `ℙ s` has to be notation for the real-valued finite measure applied to `s`, or `ℙ` has to be notation for the real-valued measure, but it then cannot be used to write `ℙ[X]` for example.

I will leave that out for now and look at what we might have about that real-valued version of a finite measure.

Reply:
I agree that often `ennreal` values for probabilities are not fun. Currently `measure_theory.finite_measure.has_coe_to_fun` and `measure_theory.probability_measure.has_coe_to_fun` yield the underlying measures of sets composed with `ennreal.to_nnreal`, but there is no API and this is therefore largely untested.

One possibility would be to try something similar to `dist` vs `nndist`. For example one could have a typeclass `has_nnmeasure`, instantiated either under the assumption `[is_finite_measure µ]` or for the types `finite_measure` and `probability_measure`. Notation should be chosen anyway. And it would require some API, but `nndist` has worked well for me recently.

I am not suggesting anything for this PR, just commenting on the status of "real-valued version of a finite measure".